packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -81,11 +88,18 @@ class Select extends Foundation<SelectHandledProps, SelectUnhandledProps, Select
             initialSelection = initialSelection.slice(0, 1);
         }
 
+        const validOptions: React.ReactNode[] = this.getValidOptions();
+
         this.state = {
             selectedItems: initialSelection,
             value: this.getValueFromSelection(initialSelection),
             displayString: this.getFormattedDisplayString(initialSelection),
             isMenuOpen: this.validateMenuState(false),
+            selectedItemIndex: this.getSelectedItemPosInSet(
+                validOptions,
+                initialSelection
+            ),
+            selectableItemCount: validOptions.length,
```

Comment:
Are we testing to see if these items in state will fall out of sync when props are updated, such as adding/removing children? If not can we add a task to address this.

A possible scenario to illustrate this point:
The select is instantiated with no items as children and is disabled, the user makes a change somewhere else in the UI which prompts a backend service to fetch data and items are populated in the select.

Reply:
Not yet.

https://github.com/microsoft/fast-dna/issues/2278